File: oop/labaratories/labaratory-work-1/salad.cpp
#include "salad.h"

Dressing Salad::GetDressing() const
{
	return _dressing;
}

void Salad::SetDressing(Dressing dressing)
{
	_dressing = dressing;
}

void Salad::ChangeDressing()
{
	_dressing = _dressing == Dressing::OliveOil ? Dressing::Mayonnaise : Dressing::OliveOil;
}

void Salad::TossWithDressing()
{
	std::cout << "Tossing the salad with " << GetDressing() << "..." << std::endl;
}

void Salad::DisplayInfo()
{
	std::cout << "Salad: " << GetName() << ", "
		<< GetDressing() << " dressing, "
		<< std::to_string(GetWeight()) << "g, $"
		<< std::fixed << std::setprecision(2) << CalcFullPrice()
		<< std::endl;
}

double Salad::CalcFullPrice()
{
	return _dressing == OliveOil ? GetPrice() * 1.4 : GetPrice() * 1.3;
}

File: oop/labaratories/labaratory-work-2/Models/Salad.cs
namespace labaratory_work_2.Models
{
    enum Dressing
    {
        OliveOil,
        Mayonnaise
    };

    public class Salad : Dish
    {
        public Dressing Dressing { get; set; } = Dressing.OliveOil;

        public Salad() : base()
        {
        }

        public Salad(string name) : base(name)
        {
        }

        public Salad(string name, int weight, double price)
            : base(name, weight, price)
        {
        }

        public override void DisplayInfo()
        {
            return $"Salad: {Name}, {Dressing.ToString()} dressing, {Weight} g, ${CalcFullPrice()}";
        }

        public override double CalcFullPrice()
        {
            return Dressing == Dressing.OliveOil ? Price * 1.4 : Price * 1.3;
        }

        public void ChangeDressing()
        {
            Dressing = Dressing == Dressing.OliveOil ? Dressing.Mayonnaise : Dressing.OliveOil;
        }

        public string TossWithDressing()
        {
            return $"Tossing the salad with {Dressing.ToString()}";
        }
    }
}
